File: android/src/main/java/com/guichaguri/trackplayer/model/State.kt
package com.guichaguri.trackplayer.model

enum class State(val value: Int) {
    //TODO: Replace with ordinal? Start from 0?
    None(1),
    Ready(2),
    Playing(3),
    Paused(4),
    Stopped(5),
    Buffering(6),
    Connecting(7)
}

File: ios/RNTrackPlayer/Models/State.swift
//
//  State.swift
//  RNTrackPlayer
//
//  Created by David Chavez on 02.12.21.
//

import Foundation
import SwiftAudioEx

enum State: String {
    case none, ready, playing, paused, stopped, buffering, connecting

    static func fromPlayerState(state: AVPlayerWrapperState) -> State {
        switch state {
        case .paused: return .paused
        case .buffering: return .buffering
        case .idle: return .none
        case .loading: return .connecting
        case .playing: return .playing
        case .ready: return .ready
        }
    }
}
